src/service/extensions-impl.js

```diff
@@ -242,32 +242,42 @@ export class Extensions {
    * @return {!Promise<!ExtensionDef>}
    */
   reloadExtension(extensionId) {
-    const attribute = isTemplateExtension(extensionId)
-      ? 'custom-template'
-      : 'custom-element';
-    // The :not is an extra prevention of recursion because it will be
-    // added to script tags that go into the code path below.
-    const oldScriptElement = this.win.document.head./*OK*/ querySelector(
-      `[${attribute}="${extensionId}"]:not([i-amphtml-inserted])`
-    );
-    devAssert(
-      oldScriptElement,
-      'Expected to find script for extension: %s',
-      extensionId
+    // Ignore inserted script elements to prevent recursion.
+    const el = this.getExtensionScript_(
+      extensionId,
+      /* includeInserted */ false
     );
-
+    devAssert(el, 'Cannot find script for extension: %s', extensionId);
     // "Disconnect" the old script element and extension record.
     const holder = this.extensions_[extensionId];
     if (holder) {
       devAssert(!holder.loaded && !holder.error);
       delete this.extensions_[extensionId];
     }
-    oldScriptElement.removeAttribute(attribute);
-    oldScriptElement.setAttribute('i-amphtml-loaded-new-version', extensionId);
-    const urlParts = parseExtensionUrl(oldScriptElement.src);
+    el.setAttribute('i-amphtml-loaded-new-version', extensionId);
+    const urlParts = parseExtensionUrl(el.src);
     return this.preloadExtension(extensionId, urlParts.extensionVersion);
   }
 
+  /**
+   * Returns the extension <script> element and attribute for the given
+   * extension ID, if it exists. Otherwise, returns null.
+   * @param {string} extensionId
+   * @param {boolean=} includeInserted If true, includes script elements that
+   *   are inserted by the runtime dynamically. Default is true.
+   * @return {?Element}
+   * @private
+   */
+  getExtensionScript_(extensionId, includeInserted = true) {
+    // Always ignore <script> elements that have a mismatched RTV.
+    const modifier =
+      ':not([i-amphtml-loaded-new-version])' +
+      (includeInserted ? '' : ':not([i-amphtml-inserted])');
+    return this.win.document.head./*OK*/ querySelector(
+      `script[src*="/${extensionId}-"]` + modifier
```

Comment:
This regex is too wide. E.g. it catches both: `amp-subscriptions` and `amp-subscriptions-google`. I'll try to go back and use an exact attribute match.